core/lib/dal/src/sync_dal.rs

```diff
@@ -53,35 +53,44 @@ impl SyncDal<'_, '_> {
             FROM
                 miniblocks
             WHERE
-                miniblocks.number = $1
+                miniblocks.number BETWEEN $1 AND $2
             "#,
-            i64::from(block_number.0)
+            i64::from(numbers.start.0),
+            i64::from(numbers.end.0 - 1),
```

Comment:
I'm not an expert in SQL, but AFAIU `BETWEEN` operator is not ordered (e.g. arguments can come in either order), and so it looks like this query will work incorrectly if called with an empty range. Maybe bail if `numbers.is_empty()`?

Reply:
thx, good catch